### PR TITLE
[subnet-visualizer] Allow to click on used ip to change them

### LINKED ISSUE #101: [subnet-visualizer] Allow to click on used ip to change them
The new subnet visualizer allows adding IP to a subnet but not changing existing ones.

We should change it to allow this.

CC: @pawelplsi

### PR DESCRIPTION
**Pull request by nepython**
Closes #101

### CODE REVIEW

**Comment by nepython** [comment]
@nemesisdesign, can you please review this :smile:

**Review by atb00ker — COMMENTED** [review]
Is it me or does it get stuck on "Closing popup" on firefox when I try to change an existing IP's information?

**Review by atb00ker — COMMENTED** [review]
Also, `?_to_field=id&_popup=1&ip_address=<ip>&subnet=<uuid>` doesn't really serve us, can it be removed from the URL without adding a lot more code? (otherwise, it's all the same)

Edit: `_popup=1&amp;` might be useful! :smile:

File: django_ipam/base/admin.py [hunk 1]
```diff
@@ -55,12 +58,23 @@ def change_view(self, request, object_id, form_url='', extra_context=None):
...
+
+        # Storing UUID corresponding to respective IP address in a dictionary
+        ip_id_list = IpAddress.objects.annotate(str_id=Cast('id', output_field=TextField())) \
+                                      .values_list('ip_address', 'str_id')
+
+        # Converting UUIdField to String and then modifying to convert back to uuid form
+        ip_id_list = OrderedDict(ip_id_list)
+        ip_uuid = {}
+        for ip_addr, Ip in ip_id_list.items():
+            ip_uuid[ip_addr] = Ip[0:8] + '-' + Ip[8:12] + '-' + Ip[12:16] + '-' + Ip[16:20] + '-' + Ip[20:]
```
Comment (atb00ker) [inline django_ipam/base/admin.py]:
```suggestion
            ip_uuid[ip_addr] = {0}-{1}-{2}-{3}.format(Ip[0:8], Ip[8:12],  Ip[12:16], Ip[16:20], Ip[20:])
```

File: django_ipam/base/admin.py [hunk 2]
```diff
@@ -55,12 +58,23 @@ def change_view(self, request, object_id, form_url='', extra_context=None):
             collection_depth += 1
 
         used = instance.ipaddress_set.count()
+
+        # Storing UUID corresponding to respective IP address in a dictionary
+        ip_id_list = IpAddress.objects.annotate(str_id=Cast('id', output_field=TextField())) \
```
Comment (atb00ker) [inline django_ipam/base/admin.py]: Can't we filter for current subnet?

**Comment by nepython** [comment]
> Is it me or does it get stuck on "Closing popup" on firefox when I try to change an existing IP's information?

Let me check this out.

**Comment by nepython** [comment]
@atb00ker, I have applied necessary changes suggested by you and taken care of the popup closing issue, though it seems I will have to add new test cases for the changes. Extremely sorry for replying so late, I had my exams going on so couldn't put in much time😅.

**Review by nemesifier — COMMENTED** [review]
@nepython can you make sure the test coverage does not decrease please?

**Comment by nepython** [comment]
Sure, I am making new test cases for the same.

**Comment by nepython** [comment]
@nemesisdesign, can you please review this?:smile:

**Review by atb00ker — APPROVED** [review]
LGTM, thanks! :+1:

**Comment by atb00ker** [comment]
P.S: f-strings joined the party in Python 3.6 not 3.7 as I thought earlier, so I think we can use it everywhere! :smile:

**Review by nemesifier — CHANGES_REQUESTED** [review]
Great, please also remove

```css
cursor: default;
```

From `.subnet-visual a.used` in the CSS so the cursor shows the normal icon when hovering on the links.

**Comment by nepython** [comment]
Done :smiley:

**Review by nemesifier — APPROVED** [review]

**Review by nemesifier — COMMENTED** [review]
Great, could you please rebase on the current master and resolve the conflict?

**Comment by nepython** [comment]
Done

**Comment by nepython** [comment]
@nemesisdesign, can you please review this :smiley:

**Review by nemesifier — APPROVED** [review]
Great work @nepython. Now the goal should be that in your next patches you can get to this point with a lot less interaction and review on our side. That's why you should keep training.

**Comment by nepython** [comment]
> Great work @nepython. Now the goal should be that in your next patches you can get to this point with a lot less interaction and review on our side. That's why you should keep training.

Sure, @nemesisdesign will try to achieve that
:)

**Pull request merged**